tcli-bepinex-installer/src/main.rs

```diff
@@ -279,22 +313,72 @@ fn uninstall_mod(bep_dir: PathBuf, name: String) -> Result<()> {
     Ok(())
 }
 
-fn output_instructions(bep_dir: PathBuf, platform: Option<String>) {
-    if platform.as_ref().map(|p| p == "windows").unwrap_or(true) {
-        let drive_prefix = match platform {
-            Some(_) => "Z:",
-            None => ""
-        };
-
-        println!("ARGUMENTS:--doorstop-enable true --doorstop-target {}{}", drive_prefix, bep_dir.join("BepInEx").join("core").join("BepInEx.Preloader.dll").to_string_lossy().replace('/', "\\"));
-        println!("WINEDLLOVERRIDE:winhttp")
+fn output_instructions(game_dir: PathBuf, bep_dir: PathBuf, platform: GamePlatform) {
+    let drive_prefix = if matches!(platform, GamePlatform::Proton) {
+        "Z:\\"
     } else {
-        eprintln!("native linux not implmented");
+        ""
+    };
+
+    let bep_preloader_dll = bep_dir
+        .join("BepInEx")
+        .join("core")
+        .join("BepInEx.Preloader.dll")
+        .to_string_lossy()
+        .into_owned();
+
+    match platform {
+        GamePlatform::Windows | GamePlatform::Proton => {
+            println!(
+                "ARGUMENTS:--doorstop-enable true --doorstop-target {}{}",
+                drive_prefix,
+                bep_preloader_dll.replace('/', "\\")
+            );
+            println!("WINEDLLOVERRIDE:winhttp");
+        }
+        GamePlatform::Linux => {
+            println!("ENVIRONMENT:DOORSTOP_ENABLE=TRUE");
+            println!("ENVIRONMENT:DOORSTOP_INVOKE_DLL_PATH={}", bep_preloader_dll);
+            println!(
+                "ENVIRONMENT:DOORSTOP_CORLIB_OVERRIDE_PATH={}",
+                game_dir.join("unstripped_corlib").to_string_lossy()
+            );
+
+            let mut ld_library = OsString::from(game_dir.join("doorstop_libs"));
+            match env::var_os("LD_LIBRARY") {
+                Some(orig) => {
+                    ld_library.push(":");
+                    ld_library.push(orig);
+                }
+                None => {}
+            }
+
+            println!("ENVIRONMENT:LD_LIBRARY={}", ld_library.to_string_lossy());
+
+            let mut ld_preload = OsString::from(game_dir.join("doorstop_libs").join({
+                if cfg!(target_arch = "x86_64") {
+                    "libdoorstop_x64.so"
+                } else {
+                    "libdoorstop_x86.so"
+                }
+            }));
```

Comment:
Make sure to leave a note describing this limitation so it can be resolved sometime later.

Reply:
Added in b34933a2a02afbd03215083d0231a894cb1a4a62